Tests/ApolloInternalTestHelpers/MockURLProtocol.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+
+public class MockURLProtocol<RequestProvider: MockRequestProvider>: URLProtocol {
+  
+  override class public func canInit(with request: URLRequest) -> Bool {
+    return true
+  }
+  
+  override class public func canonicalRequest(for request: URLRequest) -> URLRequest {
+    return request
+  }
+  
+  override public func startLoading() {
+    guard let url = self.request.url,
+          let handler = RequestProvider.requestHandlers[url] else {
+      fatalError("No URL available for URLRequest.")
```

Comment:
Isn't it the handler rather than the URL that's likely to be missing?

Reply:
Ah yea nice catch I need to re-word that error message